### PR TITLE
feat: mgmt plane release tool update

### PR DESCRIPTION
**Pull request by tadelesh**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

1. change second generate cmd to version replacement for update release
2. refine changelog file content set based on new changelog format
3. add refresh-v2 cmd for convenience
4. add package config support for muliti-package rp onboard

### CODE REVIEW

File: eng/tools/generator/cmd/v2/release/releaseCmd.go [hunk 1]
```diff
@@ -99,49 +98,20 @@ type commandContext struct {
...
+	specCommitHash, err := common.GetSpecCommit(specRepoParam)
+	if err != nil {
+		return err
 	}
 
 	log.Printf("Release generation for rp: %s, namespace: %s", c.rpName, c.namespaceName)
 	generateCtx := common.GenerateContext{
-		SDKPath:        sdkRepo.Root(),
-		SDKRepo:        &sdkRepo,
+		SDKPath:        (*sdkRepo).Root(),
```
Comment (ArcturusZhang) [inline eng/tools/generator/cmd/v2/release/releaseCmd.go:113]: Why we have to dereference this variable? this looks weird

File: eng/tools/generator/cmd/v2/common/workspaceProcessor.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/eng/tools/generator/repo"
+)
+
+var (
+	commitIDRegex = regexp.MustCompile("^[0-9a-f]{40}$")
+)
+
+func GetSDKRepo(sdkRepoParam, sdkRepoURL string) (*repo.SDKRepository, error) {
```
Comment (ArcturusZhang) [inline eng/tools/generator/cmd/v2/common/workspaceProcessor.go:18]: If I did not remember wrong, `repo.SDKRepository` is an interface, therefore we did not need to return pointer here

**Review by ArcturusZhang — COMMENTED** [review]
Some minor comments

File: eng/tools/generator/cmd/v2/release/releaseCmd.go (hunk 1 above)
Comment (tadelesh) [reply eng/tools/generator/cmd/v2/release/releaseCmd.go]: Changed.

File: eng/tools/generator/cmd/v2/common/workspaceProcessor.go (hunk 2 above)
Comment (tadelesh) [reply eng/tools/generator/cmd/v2/common/workspaceProcessor.go]: Changed.

File: eng/tools/generator/template/rpName/packageName/autorest.md.tpl [hunk 3]
```diff
@@ -9,4 +9,5 @@ require:
 - https://github.com/Azure/azure-rest-api-specs/blob/{{commitID}}/specification/{{rpName}}/resource-manager/readme.go.md
 license-header: MICROSOFT_MIT_NO_VERSION
 module-version: 0.1.0
+{{packageConfig}}
```
Comment (ArcturusZhang) [inline eng/tools/generator/template/rpName/packageName/autorest.md.tpl:12]: What does this mean?

**Review by ArcturusZhang — APPROVED** [review]

File: eng/tools/generator/template/rpName/packageName/autorest.md.tpl (hunk 3 above)
Comment (tadelesh) [reply eng/tools/generator/template/rpName/packageName/autorest.md.tpl:12]: I have added a config param to release-v2 for multi-package RP's onboard which can be used to add config like `package-singleservers: true`. Also, package config will be extracted from swagger config yaml annotation for SDK automation onboard.

**Pull request merged**